src/libbluechi/cli/command.c

```diff
@@ -0,0 +1,141 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "command.h"
+
+#include "libbluechi/common/string-util.h"
+
+Command *new_command() {
+        _cleanup_command_ Command *command = malloc0(sizeof(Command));
+        if (command == NULL) {
+                return NULL;
+        }
+
+        command->ref_count = 1;
+        command->op = NULL;
+        command->opargc = 0;
+        command->opargv = NULL;
+        command->is_help = false;
+
+        LIST_HEAD_INIT(command->command_options);
+
+        return steal_pointer(&command);
+}
+
+Command *command_ref(Command *command) {
+        command->ref_count++;
+        return command;
+}
+
+void command_unref(Command *command) {
+        command->ref_count--;
+        if (command->ref_count != 0) {
+                return;
+        }
+
+        CommandOption *option = NULL;
+        CommandOption *next_option = NULL;
+        LIST_FOREACH_SAFE(options, option, next_option, command->command_options) {
+                command_option_free(option);
+        }
+
+        free_and_null(command);
+}
+
+int command_execute(Command *command, void *userdata) {
+        const Method *method = command->method;
+        if (command->is_help) {
+                method->usage();
```

Comment:
Shouldn't you return after calling `usage`?

Reply:
Right